Automattic/jetpack
# webpack-config: Configure minifier to preserve translator comments

#### Changes proposed in this Pull Request:

This drops use of calypso-build's terser configuration in favor of one
that preserves the translator comments (and doesn't keep the various
licensing comments inline). It also simplifies the calculation of
terser's `ecma` option to match how terser actually uses it (e.g. terser
doesn't do anything different for "2015" versus "2019").
    
As for css-minimizer-webpack-plugin, this updates it from version 1.3.0
(used by calypso-build) to 3.1.1, which brings a new minification
technique that sorts the declarations for better compression.

This also fixes some code in Jetpack where the minifier loses translator
comments.

#### Jetpack product discussion


Fixes #16549

PT: p9dueE-3MG-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Check the built JS files for the presence of "translators" comments.
* Maybe check various Jetpack JS UIs to make sure the sorting of CSS declarations doesn't somehow break something.
